benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -770,10 +641,172 @@ void GraphicsBenchmarkApp::DrawExtraInfo()
     ImGui::NextColumn();
 }
 
+void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainPtr swapchain, uint32_t imageIndex)
+{
+    PPX_CHECKED_CALL(frame.cmd->Begin());
+
+    // Write start timestamp
+    frame.cmd->WriteTimestamp(frame.timestampQuery, grfx::PIPELINE_STAGE_TOP_OF_PIPE_BIT, /* queryIndex = */ 0);
+
+    frame.cmd->SetScissors(GetScissor());
+    frame.cmd->SetViewports(GetViewport());
+
+    grfx::RenderPassPtr currentRenderPass = GetRenderpass(swapchain, imageIndex);
+
+    // Transition image layout PRESENT->RENDER before the first renderpass
+    frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_RENDER_TARGET);
+
+    // Record renderpass for scene
+    frame.cmd->BeginRenderPass(currentRenderPass);
+    RecordCommandBufferSkybox(frame);
+    RecordCommandBufferSpheres(frame);
+    frame.cmd->EndRenderPass();
+
+    // Record renderpass(es) for fullscreen quads
+    uint32_t quadsCount       = pFullscreenQuadsCount->GetValue();
+    bool     singleRenderpass = pFullscreenQuadsSingleRenderpass->GetValue();
+    if (quadsCount > 0) {
+        frame.cmd->BindGraphicsPipeline(mFullscreenQuads.pipeline);
+        frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
+
+        // Begin the first renderpass
+        currentRenderPass = GetRenderpass(swapchain, imageIndex);
```

Comment:
Not sure if I am missing something here. How is the new value different from what's already in currentRenderPass assigned on line 654?

Reply:
Thanks for explaining this offline, I've removed this and also `GraphicsBenchmarkApp::GetRenderpass()`